examples_utils/benchmarks/environment_utils.py

```diff
@@ -56,18 +65,29 @@ def check_env(args: argparse.Namespace, benchmark_name: str, cmd: str):
     if ("PARTITION" in os.environ) and ("IPUOF_VIPU_API_PARTITION_ID" not in os.environ):
         os.environ["IPUOF_VIPU_API_PARTITION_ID"] = os.environ["PARTITION"]
 
-    # Check if any of the poprun env vars are required but not set
-    missing_poprun_vars = [
-        env_var for env_var in POPRUN_VARS.keys() if f"${env_var}" in cmd and os.getenv(env_var) is None
-    ]
-    if missing_poprun_vars:
-        err = (f"{len(missing_poprun_vars)} environment variables are needed by "
-               f"command {benchmark_name} but are not defined: "
-               f"{missing_poprun_vars}. Hints: \n")
-        err += "".join([f"\n\t{missing} : {POPRUN_VARS[missing]}" for missing in missing_poprun_vars])
-
-        logger.error(err)
-        raise EnvironmentError(err)
+    # if submitting on slurm, these environment variables are ignored
+    if not args.submit_on_slurm:
+        # Check if any of the poprun env vars are required but not set
+        missing_poprun_vars = [
+            env_var for env_var in POPRUN_VARS.keys() if f"${env_var}" in cmd and os.getenv(env_var) is None
+        ]
+        if missing_poprun_vars:
+            err = (f"{len(missing_poprun_vars)} environment variables are needed by "
+                   f"command {benchmark_name} but are not defined: "
+                   f"{missing_poprun_vars}. Hints: \n")
+            err += "".join([f"\n\t{missing} : {POPRUN_VARS[missing]}" for missing in missing_poprun_vars])
+
+            logger.error(err)
+            raise EnvironmentError(err)
+
+    if args.submit_on_slurm:
+        for k, v in SLURM_ENV_VARS.items():
+            if k not in os.environ:
+                warn_msg = F"{k}: {v['help']} has not been set. Falling back to the default value of: {v['default']}."
+                logger.warn(warn_msg)
+                os.environ[k] = v["default"]
+
+    # TODO: Investigate working of wandb and awscli on SLURM
```

Comment:
We can ignore AWSCLI, its going to be way too compilcated to get authentication to work tbh. For uploading, it will need authentication via aws-mfa which is a pain to setup locally, let alone remote. For downloading, we wont need this for convergence tests, since we always run phase 1 onwards or manually can setup checkpoints in the nethome shared drive. 

Wandb yes though, should be able to just copy across the two env vars, or maybe the login details from .netrc folder? I think its easy anyway.

Reply:
I don't think ther eis any work to do for this actually. All uploading/downloading will be done on the bastion host, so the user will only need to make sure it works there

Reply:
In which case the existing code for error handling should be informative enough I guess.